comp/otelcol/otlp/components/exporter/datadogexporter/factory.go

```diff
@@ -160,12 +181,34 @@ func checkAndCastConfig(c component.Config, logger *zap.Logger) *Config {
 
 // createTracesExporter creates a trace exporter based on this config.
 func (f *factory) createTracesExporter(
-	_ context.Context,
-	_ exporter.CreateSettings,
-	_ component.Config,
+	ctx context.Context,
+	set exporter.CreateSettings,
+	c component.Config,
 ) (exporter.Traces, error) {
-	// TODO implement
-	return nil, nil
+	cfg := checkAndCastConfig(c, set.TelemetrySettings.Logger)
+
+	err := f.setupTraceAgentCmp(set.TelemetrySettings)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build attributes translator: %w", err)
```

Comment:
nit: this error message seems to carry some implicit knowledge from the `setupTraceAgentCmp` implementation. We may have other errors returned in the future and this log would be deceptive.

Reply:
Good point, fixed the message here